// rc-touchable/lib/PressEvent.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.shouldFirePress = shouldFirePress;

var _objectAssign = require('object-assign');

var _objectAssign2 = _interopRequireDefault(_objectAssign);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

function PressEvent(nativeEvent) {
    var _this = this;

    this.nativeEvent = nativeEvent;
    ['type', 'currentTarget', 'target', 'touches', 'changedTouches'].forEach(function (m) {
        _this[m] = nativeEvent[m];
    });
    if (!nativeEvent.$longPressSeq) {
        nativeEvent.$longPressSeq = 1;
    } else {
        nativeEvent.$longPressSeq += 1;
    }
    this.$longPressSeq = nativeEvent.$longPressSeq;
}
(0, _objectAssign2['default'])(PressEvent.prototype, {
    preventDefault: function preventDefault() {
        this.nativeEvent.preventDefault();
    },
    stopPropagation: function stopPropagation() {
        var nativeEvent = this.nativeEvent,
            $longPressSeq = this.$longPressSeq;

        if (nativeEvent.$longPressStopSeq) {
            return;
        }
        nativeEvent.$longPressStopSeq = $longPressSeq;
    }
});
function shouldFirePress(e) {
    var nativeEvent = e.nativeEvent,
        $longPressSeq = e.$longPressSeq;

    if (!nativeEvent.$longPressStopSeq) {
        return true;
    }
    return nativeEvent.$longPressStopSeq >= $longPressSeq;
}
exports['default'] = PressEvent;
