// rc-touchable/es/PressEvent.js
import assign from 'object-assign';
function PressEvent(nativeEvent) {
    var _this = this;

    this.nativeEvent = nativeEvent;
    ['type', 'currentTarget', 'target', 'touches', 'changedTouches'].forEach(function (m) {
        _this[m] = nativeEvent[m];
    });
    if (!nativeEvent.$longPressSeq) {
        nativeEvent.$longPressSeq = 1;
    } else {
        nativeEvent.$longPressSeq += 1;
    }
    this.$longPressSeq = nativeEvent.$longPressSeq;
}
assign(PressEvent.prototype, {
    preventDefault: function preventDefault() {
        this.nativeEvent.preventDefault();
    },
    stopPropagation: function stopPropagation() {
        var nativeEvent = this.nativeEvent,
            $longPressSeq = this.$longPressSeq;

        if (nativeEvent.$longPressStopSeq) {
            return;
        }
        nativeEvent.$longPressStopSeq = $longPressSeq;
    }
});
export function shouldFirePress(e) {
    var nativeEvent = e.nativeEvent,
        $longPressSeq = e.$longPressSeq;

    if (!nativeEvent.$longPressStopSeq) {
        return true;
    }
    return nativeEvent.$longPressStopSeq >= $longPressSeq;
}
export default PressEvent;
